cpp/marc_tools/marc_augmentor.cc

```diff
@@ -373,6 +386,39 @@ bool ReplaceField(MARC::Record * const record, const MARC::Tag &tag, const char
 }
 
 
+bool ReplaceFieldRegex(MARC::Record * const record, const MARC::Tag &tag, const char subfield_code, const std::string &replacement_regex,
+                       CompiledPattern * const condition = nullptr) {
+    if (condition != nullptr) {
+        if (not condition->matched(*record))
+            return false;
+    }
+
+    bool replaced_at_least_one(false);
+    for (auto &field : *record) {
+        if (field.getTag() != tag)
+            continue;
+
+        MARC::Subfields subfields(field.getContents());
+        if (subfields.hasSubfield(subfield_code))
```

Comment:
Subfields::Subfields() has a constructor with string but this will garble up things, if the contents of the have fields have no subfields as it is the case with control fields.